### PR TITLE
add prepared statements and CAA records to the upgrade info

### PR DESCRIPTION
**Pull request by klaus3000**

### CODE REVIEW

File: docs/markdown/authoritative/upgrading.md [hunk 1]
```diff
@@ -52,3 +53,9 @@ The API path has changed to `/api/v1`.
 
 Incompatible change: `SOA-EDIT-API` now follows `SOA-EDIT-DNSUPDATE` instead of `SOA-EDIT` (incl. the fact that it now has a default value of `DEFAULT`).
 You must update your existing `SOA-EDIT-API` metadata (set `SOA-EDIT` to your previous `SOA-EDIT-API` value, and `SOA-EDIT-API` to `SOA-EDIT` to keep the old behaviour).
+
+## Ressource Record Changes
```
Comment (pieterlexis) [inline docs/markdown/authoritative/upgrading.md]: s/Ressource/Resource/

File: docs/markdown/authoritative/upgrading.md [hunk 2]
```diff
@@ -52,3 +53,9 @@ The API path has changed to `/api/v1`.
...
+Since PowerDNS 4.0.0 the CAA ressource record (type 257) is supported. In PowerDNS 3.4.x type 257 was used for a proprietary MBOXFW ressource record, which
```
Comment (pieterlexis) [inline docs/markdown/authoritative/upgrading.md]: s/ressource/resource/
and
s/In PowerDNS 3.4.x/Before PowerDNS 4.0.0/

File: docs/markdown/authoritative/upgrading.md [hunk 3]
```diff
@@ -52,3 +53,9 @@ The API path has changed to `/api/v1`.
 
 Incompatible change: `SOA-EDIT-API` now follows `SOA-EDIT-DNSUPDATE` instead of `SOA-EDIT` (incl. the fact that it now has a default value of `DEFAULT`).
 You must update your existing `SOA-EDIT-API` metadata (set `SOA-EDIT` to your previous `SOA-EDIT-API` value, and `SOA-EDIT-API` to `SOA-EDIT` to keep the old behaviour).
+
+## Ressource Record Changes
+Since PowerDNS 4.0.0 the CAA ressource record (type 257) is supported. In PowerDNS 3.4.x type 257 was used for a proprietary MBOXFW ressource record, which 
+was removed from PowerDNS 4.0. Hence, if you used CAA records with 3.4.x (stored in the DB with wrong type=MBOXFW but worked fined) and upgrade to 4.0,
```
Comment (pieterlexis) [inline docs/markdown/authoritative/upgrading.md]: s/fined/fine

File: docs/markdown/authoritative/upgrading.md [hunk 4]
```diff
@@ -52,3 +53,9 @@ The API path has changed to `/api/v1`.
...
+PowerDNS will fail to parse this records and will throw an exception on all queries for a label with MBOXFW records. Thus, make sure to clean up the
+records in the DB.
```
Comment (pieterlexis) [inline docs/markdown/authoritative/upgrading.md]: Could use a mention of running `pdnsutil check-all-zones` on upgrade.

Comment (klaus3000) [inline docs/markdown/authoritative/upgrading.md]: `pdnsutil check-all-zones` is of no help here. It does not detect or complain about the leftover MBOXFW record.

Comment (pieterlexis) [inline docs/markdown/authoritative/upgrading.md]: Alright! Thanks.

**Comment by Habbie** [comment]
Ping

**Pull request merged**